server/command_hooks.go

```
@@ -722,3 +762,89 @@ func (p *Plugin) executeAutocompleteTest(args *model.CommandArgs) *model.Command
 		Text:         fmt.Sprintf("Executed command: " + args.Command),
 	}
 }
+
+func (p *Plugin) executeCommandPreferences(args *model.CommandArgs) *model.CommandResponse {
+	fields := strings.Fields(args.Command)
+	command := ""
+	if len(fields) == 2 {
+		command = fields[1]
+	}
+
+	switch command {
+	case help:
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         commandPreferencesHelp,
+		}
+	case "get":
+		var name string
+		if len(fields) != 3 {
+			return &model.CommandResponse{
+				ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+				Text:         "Please provide preference name",
+			}
+		}
+		name = fields[2]
+		prefs, err := p.API.GetPreferencesForUser(args.UserId)
+		if err != nil {
+			p.API.LogError("Failed to get user preferences", "err", err.Error())
+			return &model.CommandResponse{}
```

Should we return a short error message here?